libraries/AP_Compass/AP_Compass_Backend.cpp

```diff
@@ -103,6 +103,8 @@ void AP_Compass_Backend::correct_field(Vector3f &mag, uint8_t i)
         state.motor_offset = mot * _compass._thr;
     } else if (_compass._motor_comp_type == AP_COMPASS_MOT_COMP_CURRENT) {
         AP_BattMonitor &battery = AP::battery();
+        // get fresh current data
+        battery.read_for_mot_compensation();
```

Comment:
See the comment left on the review, but this is the key call that I think will result in unsafe changes.

Reply:
Removed

Reply:
Removed